doc/api/esm.md

```diff
@@ -216,6 +216,61 @@ a package would be accessible like `require('pkg')` and `import
 module entry point and legacy users could be informed of the CommonJS entry
 point path, e.g. `require('pkg/commonjs')`.
 
+## Package Exports
+
+By default, all subpaths from a package can be imported (`import 'pkg/x.js'`).
+Custom subpath aliasing and encapsulation can be provided through the
+`"exports"` field.
+
+<!-- eslint-skip -->
+```js
+// ./node_modules/es-module-package/package.json
+{
+  "exports": {
+    "./submodule": "./src/submodule.js"
+  }
+}
+```
+
+```js
+import submodule from 'es-module-package/submodule';
+// Loads ./node_modules/es-module-package/src/submodule.js
+```
+
+In addition to defining an alias, subpaths not defined by `"exports"` will
+throw when an attempt is made to import them:
+
+```js
+import submodule from 'es-module-package/private-module.js';
+// Throws - Package exports error
+```
+
+> Note: this is not a strong encapsulation as any private modules can still be
+> loaded by absolute paths.
+
+Folders can also be mapped with package exports as well:
+
+<!-- eslint-skip -->
+```js
+// ./node_modules/es-module-package/package.json
+{
+  "exports": {
+    "./features/": "./src/features/"
+  }
+}
+```
+
+```js
+import feature from 'es-module-package/features/x.js';
+// Loads ./node_modules/es-module-package/src/features/x.js
+```
+
+If a package has no exports, setting `"exports": false` can be used instead of
+`"exports": {}` to indicate the package does not intend for submodules to be
+exposed.
+This is just a convention that works because `false`, just like `{}`, has no
+iterable own properties.
```

Comment:
can we document what happens if an exports:

* starts with '../'
* starts with '/'
* is an absolute URL
* doesn't lead with './', '../', or '/'
* escapes a package boundary (I actually assume this is allowed looking at the code?)

Reply:
You mean the values/targets of an export mapping? In any case - this is something we want to dig into in a dedicated change for both keys and values. I'd be concerned of missing important cases in the noise if we try to address it in this initial PR. It's listed as one of the blockers to unflagging.

Reply:
seems fine to push off until unflagging.